File: benchmark/src/php/json_tokenizer.php
<?php
$R=20000;$seed=12345;function ri($m){global$seed;$seed=($seed*16807)%2147483647;return$seed%$m;}
$parts=[];for($i=0;$i<$R;$i++){$id=ri(100000);$tag=ri(1000);$si=ri(1000);$sf=ri(100);
 $parts[]='{"id":'.$id.',"tag":"w'.$tag.'","ok":true,"score":'.$si.'.'.$sf.'}';if($i<$R-1)$parts[]=',';}
$line='['.join('',$parts).']';$st=$ss=$num=$kw=0;$pos=0;$L=strlen($line);$start=hrtime(true);
while($pos<$L){$c=$line[$pos];
 if(strpos('{[}]:,',$c)!==false){$st++;$pos++;}
 elseif($c=='"'){$ss++;$pos++;while($pos<$L&&$line[$pos]!='"')$pos++;$pos++;}
 elseif(strpos('tfn',$c)!==false){$kw++;$pos+=4;if(strpos('fn',$c)!==false)$pos++;}
 else{$num++;while($pos<$L&&strpos(',}]',$line[$pos])===false)$pos++;}}
$end=hrtime(true);$ms=($end-$start)/1e6;echo"Result: {$st}_{$ss}_{$num}_{$kw}\n";echo"Time: ".round($ms)."ms\n";

File: benchmark/src/python/json_tokenizer.py
import time
R=20000;seed=12345
def ri(m):
 global seed;seed=(seed*16807)%2147483647;return seed%m
parts=[]
for i in range(R):
 id=ri(100000);tag=ri(1000);si=ri(1000);sf=ri(100)
 parts.append('{"id":'+str(id)+',"tag":"w'+str(tag)+'","ok":true,"score":'+str(si)+'.'+str(sf)+'}')
 if i<R-1:parts.append(',')
line='['+''.join(parts)+']'
st=ss=num=kw=0;pos=0;L=len(line)
start=time.time()*1000
while pos<L:
 c=line[pos]
 if c in'{[}]:,':st+=1;pos+=1
 elif c=='"':
  ss+=1;pos+=1
  while pos<L and line[pos]!='"':pos+=1
  pos+=1
 elif c in'tfn':
  kw+=1;pos+=4
  if c in'fn':pos+=1
 else:
  num+=1
  while pos<L and line[pos]not in',}]':pos+=1
end=time.time()*1000
print(f"Result: {st}_{ss}_{num}_{kw}")
print(f"Time: {int(end-start)}ms")